src/hotspot/share/opto/type.cpp

```diff
@@ -3681,6 +3681,24 @@ bool TypeInterfaces::singleton(void) const {
   return Type::singleton();
 }
 
+bool TypeInterfaces::has_non_array_interface() const {
```

Comment:
What about using `TypeAryPtr::_array_interfaces->contains(_interfaces);` instead?

Reply:
Almost!
```cpp
return !TypeAryPtr::_array_interfaces->contains(this);
```
Contains is about TypeInterfaces, that is set of interfaces. So I just need to check that `this` is not a sub-set of array interfaces. That should do it.

Reply:
Now I'm confused, isn't this what I proposed? I didn't check the exact syntax, I just wondered if the `TypeInterfaces::contains` method couldn't be used instead of adding a new method.

Reply:
Yes, totally! It's just a detail difference. But there is another question: whether we still want `has_non_array_interface` has a wrapper for this call with a more explicit name, or if we simply inline your suggestion on the callsite of `has_non_array_interface`. I tend toward the first, I like explicit names, and I suspect it might be useful in more than one place, but not a strong opinion.

Reply:
For now, I just replaced the implementation of `has_non_array_interface`. If one feels even keeping the method is premature factorization, I can easily inline it.